dts/bindings/serial/infineon,xmc4xxx-uart.yaml

```diff
@@ -7,3 +7,51 @@ include: uart-controller.yaml
 properties:
     reg:
       required: true
+
+    fifo-start-offset:
```

Comment:
Can't this be figured out based on how `fifo-tx-size` and `fifo-rx-size` are set?

Reply:
A FIFO is shared between two USIC channels. We could have two UARTs sharing the same FIFO, and that's why we need to set the offset. So yes, we could use a static array that tracks the FIFO usage per USIC and set the offset at runtime.

Unfortunately, the problem is more subtle because a USIC channel can also be configured as SPI, I2C,etc... Then tracking the FIFO usage is more complex. The USIC would have to be its own driver that stores the FIFO usage. UART, SPI,.. drivers would then have to reference the USIC in their config structs. 

I thought that asking the user to set the FIFO offset was reasonable. A separate USIC driver could also be added in the future if needed. Personally, I prefer the current approach but am open to other views.

Reply:
Marking `DNM` to give me chance to see about working up some code to go from one channel to the other to get its fifo size props.